service/src/main/java/greencity/service/notification/NotificationServiceImpl.java

```diff
@@ -87,20 +91,43 @@ public void notifyUnpaidOrders() {
     }
 
     private boolean checkIfUnpaidOrderNeedsNewNotification(Order order, Optional<UserNotification> lastNotification) {
-        return (lastNotification.isEmpty()
-            || (lastNotification.get().getNotificationTime().isBefore(LocalDateTime.now(clock).minusDays(7))
-                || lastNotification.get().getNotificationTime().isEqual(LocalDateTime.now(clock).minusDays(7))))
-            && (order.getOrderDate().isAfter(LocalDateTime.now(clock).minusMonths(1))
-                || order.getOrderDate().isEqual(LocalDateTime.now(clock).minusMonths(1)))
-            && (order.getOrderDate().isBefore(LocalDateTime.now(clock).minusDays(3))
-                || order.getOrderDate().isEqual(LocalDateTime.now(clock).minusDays(3)));
+        if (lastNotification.isEmpty()) {
+            LocalDate now = LocalDate.now(clock);
+            LocalDateTime after3Days = order.getOrderDate();
+            LocalDate orderDate = after3Days.toLocalDate().plusDays(3);
+            return now.isEqual(orderDate);
+        }
+        LocalDate now = LocalDate.now(clock);
+        LocalDate nextNotification = lastNotification.get().getNextNotification().toLocalDate();
+        return now.isEqual(nextNotification);
     }
 
     private UserNotification initialiseNotificationForUnpaidOrder(Order order, UserNotification userNotification) {
+        Optional<UserNotification> lastNotificationOptional = userNotificationRepository
+            .findLastNotificationByNotificationTypeAndOrderNumber(NotificationType.UNPAID_ORDER.toString(),
+                order.getId().toString());
+
+        if (lastNotificationOptional.isPresent()) {
+            UserNotification lastNotification = lastNotificationOptional.get();
+            if (Objects.isNull(lastNotification.getNextNotification())) {
+                lastNotification.setNextNotification(LocalDateTime.now(clock));
```

Comment:
why the next notification is equal to now?
Should it be` now + 3 days`?

Reply:
I forgot delete if statement on 112 line, because on database a couple of values was with null values and it was a way to fix it